packages/components/src/typings/PolymorphicGenerics.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * These were heavily inspired by "Forwarding refs for a polymorphic React
+ * component in TypeScript", written by Ben Ilegbodu.
+ *
+ * Article link: https://www.benmvp.com/blog/forwarding-refs-polymorphic-react-component-typescript/
+ */
+
```

Comment:
Is the same file that is being added on this [PR](https://github.com/vtex/faststore/pull/1543/files)?